### ISSUE TITLE
BigQuery connector: error when querying data

### ISSUE THREAD

**Issue by ItayGabbay** [prestodb/presto#16145]
Hi,
After successful configuration of the BigQuery catalog, I try to query some data from one of the tables using a simple `SELECT * FROM <bq_dataset>.<bq_table>`, but get an internal error:

```
java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.get(Optional.java:148)
	at com.facebook.presto.plugin.bigquery.BigQueryMetadata.getTableHandle(BigQueryMetadata.java:123)
	at com.facebook.presto.metadata.MetadataManager.getTableHandle(MetadataManager.java:334)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:1087)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:312)
	at com.facebook.presto.sql.tree.Table.accept(Table.java:53)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:326)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.analyzeFrom(StatementAnalyzer.java:2292)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:1355)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:312)
	at com.facebook.presto.sql.tree.QuerySpecification.accept(QuerySpecification.java:127)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:326)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:336)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:938)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:312)
	at com.facebook.presto.sql.tree.Query.accept(Query.java:94)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:326)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.analyze(StatementAnalyzer.java:304)
	at com.facebook.presto.sql.analyzer.Analyzer.analyze(Analyzer.java:82)
	at com.facebook.presto.sql.analyzer.Analyzer.analyze(Analyzer.java:74)
	at com.facebook.presto.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:183)
	at com.facebook.presto.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:94)
	at com.facebook.presto.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:768)
	at com.facebook.presto.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:117)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by ItayGabbay** [comment]
When taking a look at the code, I suspect there might be some issue with the `private Optional<TableInfo> getBigQueryTable(SchemaTableName tableName)` in the `BigQueryMetadata.java` file:

```
private Optional&lt;TableInfo&gt; getBigQueryTable(SchemaTableName tableName)    {        
    TableInfo tableInfo = bigQueryClient.getTable(TableId.of(projectId, tableName.getSchemaName(), 
    tableName.getTableName()));        
    return tableInfo == null ? Optional.of(tableInfo) : Optional.empty();    
}

```

Either way, it returns an empty table.

**Comment by hendrikfiskhjarna** [comment]
I'm getting a similar error when running the same operation. I can list the tables in the schema etc, but querying the data fails:

`java.lang.NullPointerException:` undefined
	at java.base/java.util.Objects.requireNonNull(Objects.java:208)
	at java.base/java.util.Optional.of(Optional.java:111)
	at com.facebook.presto.plugin.bigquery.BigQueryMetadata.getBigQueryTable(BigQueryMetadata.java:160)
	at com.facebook.presto.plugin.bigquery.BigQueryMetadata.getTableHandle(BigQueryMetadata.java:118)
	at com.facebook.presto.metadata.MetadataManager.getTableHandle(MetadataManager.java:334)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:1087)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:312)
	at com.facebook.presto.sql.tree.Table.accept(Table.java:53)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:326)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.analyzeFrom(StatementAnalyzer.java:2292)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:1355)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:312)
	at com.facebook.presto.sql.tree.QuerySpecification.accept(QuerySpecification.java:127)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:326)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:336)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:938)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:312)
	at com.facebook.presto.sql.tree.Query.accept(Query.java:94)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:326)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.analyze(StatementAnalyzer.java:304)
	at com.facebook.presto.sql.analyzer.Analyzer.analyze(Analyzer.java:82)
	at com.facebook.presto.sql.analyzer.Analyzer.analyze(Analyzer.java:74)
	at com.facebook.presto.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:183)
	at com.facebook.presto.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:94)
	at com.facebook.presto.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:768)
	at com.facebook.presto.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:117)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)

**Comment by rohanpednekar** [comment]
@ItayGabbay, can you share the versions you using here? 
@hendrikfiskhjarna you are facing NPE and seems a different issue, can you open a new issue with more details?

**Comment by rohanpednekar** [comment]
Hi @davidrabinowitz Do you think you can help here?

**Comment by imjalpreet** [comment]
@ItayGabbay As you pointed out there is a bug in that method. Thanks for reporting. Did you try adding a fix? If not, we can add a fix and provide you with a patch to unblock you. 

@hendrikfiskhjarna It looks like you are also getting the error from the same method which has a bug. But in your case it looks like the table you were querying did not exist in BQ due to which you got a NPE (Ideally it should have thrown a TableNotFoundException if the bug did not exist). As you said you are able to list the tables but you are facing this error when you try to query, I wanted to confirm whether the database/tables you are querying have mixed-case/upper-case names in BQ?

**Comment by ItayGabbay** [comment]
@rohanpednekar I'm using 0.253.

@imjalpreet Thanks for your answer. I didn't try to fix it, I can try. But if you can provide a quick patch that would be optimal.

**Comment by hendrikfiskhjarna** [comment]
@imjalpreet our datasets and tables are only named using lower letters and underscores.

**Comment by hendrikfiskhjarna** [comment]
@rohanpednekar new issue opened: https://github.com/prestodb/presto/issues/16178

**Comment by ItayGabbay** [comment]
Hey guys, any update about this?

**Comment by rohanpednekar** [comment]
Hi @ItayGabbay, as we have identified the bug, we will try to fix it and create the PR to close this issue. Thanks!

**Comment by highker** [comment]
fixed

**Comment by thuongle2210** [comment]
> fixed

Can you recommend me a docker images, which is fixed this error?
